synapse/handlers/typing.py

```diff
@@ -362,16 +362,11 @@ async def _recv_edu(self, origin: str, content: JsonDict) -> None:
             )
             return
 
-        domains = await self._storage_controllers.state.get_current_hosts_in_room(
-            room_id
-        )
-
-        if self.server_name in domains:
```

Comment:
could someone explain why it is safe to remove this check?

Reply:
It's lost in a resolved comment but we do check the same stuff higher up in the code:
https://github.com/matrix-org/synapse/pull/13830#discussion_r977375785

Reply:
Not the good comment sorry... 
https://github.com/matrix-org/synapse/pull/13830#discussion_r977355324

Reply:
https://github.com/matrix-org/synapse/blob/develop/synapse/handlers/typing.py#L343

Reply:
The previous check was buggy and actually tests whether the local server is in the room, rather than the remote server (#13869). ie. this PR fixes the faster room join case without introducing any new bugs.

Reply:
I can easily do so indeed once #13874 is in so I can use `get_current_hosts_in_room_or_partial_state_approximation`.

Let's go with that, I am going to dismiss Sean +1 for now so it doesn't get merged accidentally.